server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -69,8 +69,9 @@ public class LocalStorageManager extends SingleStorageManager {
   private final List<LocalStorage> localStorages;
   private final List<String> storageBasePaths;
   private final LocalStorageChecker checker;
-  private List<LocalStorage> unCorruptedStorages = Lists.newArrayList();
-  private final Set<String> corruptedStorages = Sets.newConcurrentHashSet();
+
+  // AppId -> ShuffleId -> PartitionId -> LocalStorage
+  private final Map<String, Map<Integer, Map<Integer, LocalStorage>>> partitionsOfStorage = Maps.newConcurrentMap();
```

Comment:
The mechanism of uploading local files has been removed in current codebase. And this PR is just to avoid selecting storage which has reached high watermark.

Reply:
Here https://github.com/apache/incubator-uniffle/pull/51